core/domain/story_validators.py

```diff
@@ -0,0 +1,219 @@
+# coding: utf-8
+#
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Validators for story models."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.domain import base_model_validators
+from core.domain import story_domain
+from core.domain import story_fetchers
+from core.platform import models
+import python_utils
+
+(
+    base_models, exp_models,
+    story_models, user_models
+) = models.Registry.import_models([
+    models.NAMES.base_model, models.NAMES.exploration,
+    models.NAMES.story, models.NAMES.user
+])
+
+
+class StoryModelValidator(base_model_validators.BaseModelValidator):
+    """Class for validating StoryModel."""
+
+    @classmethod
+    def _get_model_domain_object_instance(cls, item):
+        return story_fetchers.get_story_from_model(item)
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        snapshot_model_ids = [
+            '%s-%d' % (item.id, version)
+            for version in python_utils.RANGE(1, item.version + 1)]
+        return [
+            base_model_validators.ExternalModelFetcherDetails(
+                'story_commit_log_entry_ids',
+                story_models.StoryCommitLogEntryModel,
+                ['story-%s-%s'
+                 % (item.id, version) for version in python_utils.RANGE(
+                     1, item.version + 1)]),
+            base_model_validators.ExternalModelFetcherDetails(
+                'story_summary_ids',
+                story_models.StorySummaryModel, [item.id]),
+            base_model_validators.ExternalModelFetcherDetails(
+                'snapshot_metadata_ids',
+                story_models.StorySnapshotMetadataModel,
+                snapshot_model_ids),
+            base_model_validators.ExternalModelFetcherDetails(
+                'snapshot_content_ids',
+                story_models.StorySnapshotContentModel,
+                snapshot_model_ids),
+            base_model_validators.ExternalModelFetcherDetails(
+                'exploration_ids',
+                exp_models.ExplorationModel,
+                [node['exploration_id'] for node in (
```

Comment:
Clarifying prev comment, we have node['prerequisite_skill_ids'] and node['acquired_skill_ids'] as well.

Reply:
Hi @aks681 , Thanks for the comment. Could you please help me and elaborate on how I can add these as well? 
Should it be like this ?
base_model_validators.ExternalModelFetcherDetails(  
                'prerequisite_skill_ids',
                exp_models.ExplorationModel,
                [node['prerequisite_skill_id'] for node in (
                item.story_contents['nodes'])])

Reply:
@purvimisal I think it is okay to not include the checks for `prerequisite_skill_ids` and `acquired_skill_ids` in this PR.

@aks681 Hi, I think that the scope of the issue this relates to is just to move the validators, since the issue is for new contributors I wouldn't overload them with more complicated stuff, I agree with you point though so could you please file an issue for this?

Reply:
Ah ok, no problem. I'll file an issue for it